velox/type/Variant.cpp

```diff
@@ -687,6 +687,18 @@ bool variant::equalsWithEpsilon(const variant& other) const {
   if (other.isNull()) {
     return this->isNull();
   }
+  if (kind_ == TypeKind::ROW) {
```

Comment:
The epsilon comparison should only happen for FLOAT/DOUBLE values within the row. Can you enhance this check to look for only FLOAT/DOUBLE columns within the row ?

Reply:
For ROW members, the comparison will be done appropriately based on the type. In the following lines, it checks if the type is REAL or DOUBLE and uses epsilon comparator, for the rest of the types it falls back to the equals method. Just to make the test assertion code use epsilon comparator instead of value comparator, I had to make this change.